### DISCUSSION TITLE
Cannot find module or its corresponding type declarations.

### DISCUSSION THREAD

**Discussion by dushmanta05** [aws/aws-cdk#30769] [Q&A]
Hello all,
I'm trying to migrate my `aws-sdk` V2 to V3 with the help of [aws-sdk-js-codemod](https://github.com/aws/aws-sdk-js-codemod) after I ran the command to migrate, it completed it, but the import modules are showing error: Cannot find module `<module name>` or its corresponding type declarations.

In V2, I was using the SDK import like below:
```js
import AWS from "aws-sdk";

const s3 = new AWS.S3({
   accessKeyId: ID,
   secretAccessKey: SECRET,
});

const url = s3.getSignedUrl("getObject", {
   Bucket: BUCKET_NAME,
   Key: KEY_NAME,
});
```

After migrating to V3,

```js
import AWS from "aws-sdk";
import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
import { GetObjectCommand, S3 } from "@aws-sdk/client-s3";

AWS.config.update({
    accessKeyId: ID,
    secretAccessKey: SECRET,
});

const s3 = new S3({
    credentials: {
        accessKeyId: ID,
        secretAccessKey: SECRET,
    },
});

const url = await getSignedUrl(
    s3,
    new GetObjectCommand({
        Bucket: BUCKET_NAME,
        Key: Key,
    })
);
```

But I'm getting error like this:

```
I cannot find module '@aws-sdk/s3-request-presigner' or its corresponding type declarations.
I cannot find module '@aws-sdk/client-s3' or its corresponding type declarations.
```

Could anyone help me with this? I appreciate your time. Thank you.

**Comment by dil-mezzy** [comment]
Verify Installation:

Make sure you have installed the necessary v3 client libraries in your project. Run the following command in your terminal:

Bash
npm install @aws-sdk/client-s3 @aws-sdk/s3-request-presigner

If you're using yarn, use yarn add @aws-sdk/client-s3 @aws-sdk/s3-request-presigner.

Check Codemod Output:

Examine the migrated code files generated by aws-sdk-js-codemod. Look for any potential errors or incomplete changes.
Address Missing Imports:

Manually add the missing import statements for the v3 client libraries:

JavaScript
import { S3 } from "@aws-sdk/client-s3";
import { getSignedUrl } from "@aws-sdk/s3-request-presigner";

Update Configuration (if applicable):

If you're using a separate configuration file (e.g., aws-sdk.config.js), update it to use the v3 client constructors (S3.from instead of AWS.config.update).
Consider Re-running Codemod (cautiously):

Before re-running the codemod script, create a backup of your code.

If the previous steps don't resolve the issue, consider re-running aws-sdk-js-codemod only on the specific files that need migration. Make sure you're using the latest version of the codemod script to avoid potential bugs.

Refer - https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/migrating.html

**Comment by suyash-purwar** [comment]
Hey @dushmanta05 
I am running into the same issue now. Did you find any solution for this?

**Comment by rtsedenova** [comment]
most likely its stupid
I tried different ways to solve this problem but nothing worked
and then I just changed import to require and everything worked (even though for TS they recommend using import)
